### PR TITLE
[IMP] l10n_br: include electronic invoicing flow

### PR DESCRIPTION
**Pull request by vbe-odoo**
With the addition of l10n_br_edi and l10n_br_edi_sale to the previous modules, now electronic invoices for goods can be created from Odoo.

This PR includes the necessary information to use these new modules - configuration and workflows.

G-Folder with images: https://drive.google.com/drive/folders/1rCkyiYnWslzlQ7Oxb3IJC0-TeiWrQOwb?usp=drive_link

### CODE REVIEW

**Comment by vbe-odoo** [comment]
Here is the PR @samueljlieber with the modifications for the e-invoicing documentation.
Please review it, and feel free to edit the content to improve it.

If you have any questions do not hesitate to ping me.
These modules with the new e-invoicing feature is not yet released but it will be pretty soon (1 to 2 weeks)

Thank you!
cc' @ren-odoo @fvz-odoo @cas-odoo

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 1]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+   :alt: Number range invalidation selection in Odoo.
+
+.. image:: brazil/range-number-invalidation-wizard.png
+   :alt: Number range invalidation wizard in Odoo.
+
+.. note::
+   The log of the numbers that were canceled and the XML file will be recorded in the chatter of the
+   journal.
+
+Electronic documents
```
Comment (vbe-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:504]: Hey @samueljlieber my bad here! This is wrong, it should be
```suggestion
Vendor Bills
```

Can you help me with this change?

All the rest looks good!
cc' @ren-odoo @fvz-odoo

**Comment by samueljlieber** [comment]
Thank you @vbe-odoo, I've implemented [that change](https://github.com/odoo/documentation/pull/6877#discussion_r1419729699) in https://github.com/odoo/documentation/commit/293349c4855e2c1ee762ba56c0a95382f30450b1.

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 2]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+
+.. image:: brazil/invoice-cancellation.png
+   :alt: Invoice cancellation reason in Odoo.
+
+Correction letter
+~~~~~~~~~~~~~~~~~
+
+It is possible to create a correction letter linked to an electronic invoice that was validated by
+the government.
+
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:476]:
```suggestion
A correction letter ca be created and linked to an electronic invoice that was validated by
the government.

```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 3]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+This can be done in Odoo by clicking :guilabel:`Correction Letter` and adding a correction
+:guilabel:`Reason` on the pop-up that appears. If you want to send this correction reason to your
+customer via email, you can do so by selecting the :guilabel:`E-mail` checkbox.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:479]:
```suggestion
:guilabel:`Reason` on the pop-up that appears. To send this correction reason to a customer via email, activate the :guilabel:`E-mail` checkbox.
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 4]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+
+.. image:: brazil/correction-letter.png
+   :alt: Correction letter reason in Odoo.
+
+Invalidate invoice number range
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A range of numbers within series that are assigned to sales journals can be invalidated with the
+government, if those sequences are not used and are not going to be used. To do so, navigate to the
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:488]:
```suggestion
government, only if those sequences are not used **and** are not going to be used. To do so, navigate to the
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 5]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+journal select the ::menuselection:`⚙️ (gear) icon --> Invalidate Number Range (BR)`. On the
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:489]:
```suggestion
journal select the ::menuselection:`⚙️ (gear) icon --> Invalidate Number Range (BR)`. On the
```
```suggestion
journal, and click the ::menuselection:`⚙️ (gear) icon --> Invalidate Number Range (BR)`. On the
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 6]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+:guilabel:`Invalidate Number Range (BR)` wizard, add the :guilabel:`Initial Number` and
+:guilabel:`End Number` of the range that should be canceled, and state an invalidation
+:guilabel:`Reason`.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:492]:
```suggestion
:guilabel:`End Number` of the range that should be canceled, and enter an invalidation
:guilabel:`Reason`.
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 7]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+
+.. image:: brazil/range-number-invalidation.png
+   :alt: Number range invalidation selection in Odoo.
+
+.. image:: brazil/range-number-invalidation-wizard.png
+   :alt: Number range invalidation wizard in Odoo.
+
+.. note::
+   The log of the numbers that were canceled and the XML file will be recorded in the chatter of the
+   journal.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:502]:
```suggestion
   The log of the canceled numbers along with the XML file are recorded in the chatter of the
   journal.
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 8]
```diff
@@ -316,5 +371,147 @@ following ways:
...
+
+Vendor bills
+------------
+
+On the vendor bills side, when receiving an invoice from a supplier, you can encode the bill in Odoo
+by adding all the commercial information together with the same Brazilian specific information that
+is recorded on :ref:`customer invoices <brazil/electronic-documents>`.
```
Comment (larm-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:509]:
```suggestion
is recorded on the :ref:`customer invoices <brazil/electronic-documents>`.
```

**Comment by samueljlieber** [comment]
Thank you @larm-odoo for your very thorough review! I implemented your suggestions in https://github.com/odoo/documentation/commit/f6f20eb81b090573888e6a417504c12805193b6b.

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 9]
```diff
@@ -316,5 +372,150 @@ following ways:
...
+.. image:: brazil/invoice-info-needed.png
+   :alt: Invoice information needed to process an electronic invoice.
+
+.. image:: brazil/process-electronic-invoice.png
+   :alt: Process electronic invoice pop-up in Odoo.
+
+.. note::
+   All fields available on the invoice in order to issue an electronic invoice are also available on
+   the sales order, if needed. When creating the first invoice, the field :guilabel:`Document
+   Number` is displayed, allocated as  the first number to be used sequentially for subsequent
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:417]:
```suggestion
   Number` is displayed, allocated as the first number to be used sequentially for subsequent
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 10]
```diff
@@ -8,12 +8,14 @@ Brazil
...
-With the Brazilian localization you can automatically compute sales taxes for goods using AvaTax
-(Avalara) through API calls, also configure taxes for services.
+With the Brazilian localization, sales taxes can be automatically computed and electronic invoices
+(NF-e) for goods can be sent using AvaTax (Avalara) through API calls. Moreover, taxes for services
+can be configured.
 
-For the goods tax computation part, you need to configure the :ref:`contacts <brazil/contacts>`,
-:ref:`company <brazil/company>`, :ref:`products <brazil/products>`, and :ref:`create an account in
-Avatax <brazil/avatax-account>` from the Odoo general settings.
+For the goods tax computation and electronic invoicing part, you need to configure the
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:15]:
```suggestion
For the goods tax computation and electronic invoicing process, you need to configure the
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 11]
```diff
@@ -130,15 +150,10 @@ password:
...
-   email address is recommended, as you won't be able to re-use the same email address on your
-   production database.
-
 .. tip::
-   You can start using AvaTax in Odoo without creating a password and accessing the Avalara Portal;
-   for Odoo, the only requirement to start using the Avalara Tax Computation Engine is to create an
-   account from the settings page.
+   You can start using AvaTax in Odoo for tax computation **only** without creating a password and
+   accessing the Avalara portal in the Odoo database. But if you want to use the electronic invoice
+   service, you **must** access the AvaTax portal and upload your certificate there.
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:156]:
```suggestion
   accessing the Avalara portal in the Odoo database. However, in order to use the electronic invoice
   service, you **must** access the AvaTax portal and upload your certificate there.
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 12]
```diff
@@ -147,23 +162,53 @@ password:
    You can transfer API credentials. Use this only when you have already created an account in
    another Odoo instance and wish to reuse it.
 
+A1 certificate upload
+~~~~~~~~~~~~~~~~~~~~~
+
+In order to issue electronic invoices a certificate needs to be uploaded to the `AvaTax portal
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:168]:
```suggestion
In order to issue electronic invoices, a certificate needs to be uploaded to the `AvaTax portal
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 13]
```diff
@@ -147,23 +162,53 @@ password:
...
+<https://portal.avalarabrasil.com.br/Login>`_.
+
+The certificate will be synchronized with Odoo as far as in the AvaTax portal the external
+identifier number matches - without special characters - with the CNPJ number and in Odoo the
+identification number (CNPJ) matches with the CNPJ in AvaTax.
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:173]:
```suggestion
The certificate will be synchronized with Odoo, as long as the external identifier number in the AvaTax portal matches - without special characters - with the CNPJ number, and the identification number (CNPJ) in Odoo matches with the CNPJ in AvaTax.
```
I believe I was able to get the gist of this, but if I am wrong feel free to change it.

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 14]
```diff
@@ -231,8 +276,14 @@ Before using the integration, specify the following information on the contact:
    - Select the :guilabel:`Company` option for a contact with a tax ID (CNPJ), or check
      :guilabel:`Individual` for a contact with a CPF.
    - :guilabel:`Name`
-   - :guilabel:`Address`: :guilabel:`Zip Code` is a required field to compute taxes properly.
-   - :guilabel:`Tax ID` or :guilabel:`CPF`: Use CPF for individuals and Tax ID (CNPJ) for companies
+   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
+     :guilabel:`Country`)
+
+     - In the :guilabel:`Street` field, enter the street, number, and any extra address information
+     - In the :guilabel:`Street 2` field enter the neighborhood
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:283]:
```suggestion
     - In the :guilabel:`Street 2` field, enter the neighborhood
```

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 15]
```diff
@@ -231,8 +276,14 @@ Before using the integration, specify the following information on the contact:
...
+
+   - :guilabel:`Identification Number` (:guilabel:`CNPJ`, :guilabel:`CPF`)
+   - :guilabel:`Tax ID` (associated with the identification type)
    - :guilabel:`IE`: state tax identification number
    - :guilabel:`IM`: municipa tax identification number
```
Comment (jero-odoo) [inline content/applications/finance/fiscal_localizations/brazil.rst:288]:
```suggestion
   - :guilabel:`IM`: municipal tax identification number
```

**Review by jero-odoo — APPROVED** [review]
Hey @samueljlieber nice work, just a few comments (mostly suggestions). Let me know if you have any questions 👍

**Review by larm-odoo — APPROVED** [review]
Looks great- approved!

**Review by samueljlieber — APPROVED** [review]
Thank you @jero-odoo for your review! I implemented your suggestions in https://github.com/odoo/documentation/commit/4d2ab9c690f673be58b6a4670d403ce37540abcf. 

@vbe-odoo let me know the feature this PR addresses was merged on Monday the 11th, so @StraubCreative this PR is good to go 👍

**Comment by vbe-odoo** [comment]
> Thank you @jero-odoo for your review! I implemented your suggestions in [4d2ab9c](https://github.com/odoo/documentation/commit/4d2ab9c690f673be58b6a4670d403ce37540abcf).
> 
> @vbe-odoo let me know the feature this PR addresses was merged on Monday the 11th, so @StraubCreative this PR is good to go 👍

Hello! 
I've checked the last changes and it looks good.
Thanks for the changes and the review!

Do you think this can be merged soon? As the feature is already deployed.

Thanks!
cc' @ren-odoo @fvz-odoo

**Comment by StraubCreative** [comment]
> Do you think this can be merged soon? As the feature is already deployed.

Once we get a ✅ from the @odoo/localizations-doc-review team we can merge. They were just pinged here and I will drop in doc channel as well. Hopefully by early next week we'll be g2g!

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 16]
```diff
@@ -275,14 +327,18 @@ order or an invoice.
...
 .. warning::
    Please note that each API call incurs a cost. Be mindful of the actions that trigger these calls
    to manage costs effectively.
 
-Tax calculations on quotation / sales orders
---------------------------------------------
+Tax computation
+---------------
+
+Tax calculations on quotations / sales orders
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/brazil.rst:340]: Tax calculations on quotations and sales orders

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 17]
```diff
@@ -275,14 +327,18 @@ order or an invoice.
 Workflows
 =========
 
-This section provides an overview of the actions that trigger API calls for tax computation.
+This section provides an overview of the actions that trigger API calls for tax computation, and how
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/brazil.rst:330]: you might want to explain what API calls in a couple of words or only add a link to an explanation.

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 18]
```diff
@@ -316,5 +372,150 @@ following ways:
...
+
+There are some other optional fields that depend on the nature of the transaction. These fields are
+not required, so no errors will appear from the government if these optional fields are not
+populated for most cases:
+
+- :guilabel:`Freight Model` determines how the goods are planned to be transported - domestic.
+- :guilabel:`Transporter Brazil` determines who is doing the transportation.
+
+.. image:: brazil/invoice-info-needed.png
+   :alt: Invoice information needed to process an electronic invoice.
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/brazil.rst:409]: no red highlights

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 19]
```diff
@@ -316,5 +372,150 @@ following ways:
...
+
+.. image:: brazil/process-electronic-invoice.png
+   :alt: Process electronic invoice pop-up in Odoo.
```
Comment (chiaraprattico) [inline content/applications/finance/fiscal_localizations/brazil.rst:412]: no red highlight needed

**Review by chiaraprattico — CHANGES_REQUESTED** [review]
Nice one, thanks. I added my review, we usually don't have highlights on screenshots so I noted that but it's up to you if you think that's useful but we try to be constant. Same for punctuation in bullet points.

File: content/applications/finance/fiscal_localizations/brazil.rst [hunk 20]
```diff
@@ -256,15 +307,16 @@ Before using the integration, specify the following information on the contact:
    - :guilabel:`Main Activity Sector`: list of main activity sectors of the contact
```
[the remark this answers is not in the document]
Comment (samueljlieber) [reply content/applications/finance/fiscal_localizations/brazil.rst:307]: Updated all bullets to use periods, thank you!

**Comment by samueljlieber** [comment]
Thank you @chiaraprattico for your thorough review! I implemented all of your changes in https://github.com/odoo/documentation/commit/055ba821b5db45ebadd3bb398cd24341b4237475 except for updating the images, as I do feel the markup is helpful. That being said, this PR is ready for another look– thank you 🙏

**Comment by ren-odoo** [comment]
Hi team 

Thank you for all the hard work on this PR, most of our Brazilian partners are already very eager to know in detail the feature we have released on BR, so the sooner we can publish this documentation the better. 

Thank you.

**Comment by StraubCreative** [comment]
> Thank you @chiaraprattico for your thorough review! I implemented all of your changes in [055ba82](https://github.com/odoo/documentation/commit/055ba821b5db45ebadd3bb398cd24341b4237475) except for updating the images, as I do feel the markup is helpful. That being said, this PR is ready for another look– thank you 🙏

Since the items were addressed let's merge what we have for now so our translators can get started and so BR users can benefit from the content.

Any further issues we can address in another PR.

Thanks everyone for your hard work on this!

@robodoo r+

**Pull request closed without merging**